tests/test_validation_mismatch.py

```diff
@@ -0,0 +1,781 @@
+            obj, is_good = vjson.forecast_object_from_json(json, channel)
+            #At this point, may not be a good object if the forecast needed to use
+            #a mismatched energy channel. Check that first before determine
+            #outcome of object.
+            #If the object is good, include here
+            if not pd.isnull(obj.prediction_window_start):
+                model_objs[key].append(obj)
+                logger.debug("Created FORECAST object from json " + str(obj.source)  + ", " + key)
+                logger.debug("Prediction window start: " + str(obj.prediction_window_start))
+ 
+            if not is_good:
+                logger.warning("Note issue with creating FORECAST object from json " + str(obj.source)  + ", " + key)
+
+            #If mismatched observation and prediction energy channels
+            #enabled, then find the correct prediction energy channel
+            #to load.
+            if config_tests.do_mismatch:
+                if config_tests.mm_model in short_name:
+                    if channel == config_tests.mm_obs_energy_channel:
+                        pred_channel = config_tests.mm_pred_energy_channel
+                        obj, is_good = vjson.forecast_object_from_json(json, pred_channel)
+ 
+                        if not is_good:
+                            logger.warning("Note issue with creating FORECAST object from json " + str(obj.source)  + ", mismatch channel" + str(pred_channel))
+
+                        #skip if energy block wasn't present in json
+                        if not pd.isnull(obj.prediction_window_start):
+                            model_objs[config_tests.mm_energy_key].append(obj)
+                            logger.debug("Adding " + obj.source + " to dictionary under key " + key)
+
+
+
+
+    #Convert all_energy_channels to an array of string keys
+    for i in range(len(all_energy_channels)):
+        all_energy_channels[i] = objh.energy_channel_to_key(all_energy_channels[i])
+        
+    if config_tests.do_mismatch:
+        all_energy_channels.append(config_tests.mm_energy_key)
+    logger.debug('IN VJSON CHECKING MISMATCH')
+    logger.debug(all_energy_channels)
+    
+    del obs_jsons
+    del model_jsons
+
+    for channel in all_energy_channels:
+        logger.info("STATS: Observation objects created for : " + channel + ", " + str(len(obs_objs[channel])))
+        logger.info("STATS: Forecast objects created for : " + channel + ", " + str(len(model_objs[channel])))
+
+    return all_energy_channels, obs_objs, model_objs
+
+
+
+def utility_match_sphinx(all_energy_channels, model_names, obs_objs, model_objs):
+
+    # utility_setup_logging()
+
+    matched_sphinx, all_observed_thresholds, observed_sep_events =\
+        match.match_all_forecasts(all_energy_channels, model_names,
+            obs_objs, model_objs)
+
+    return matched_sphinx, all_observed_thresholds, observed_sep_events
+    
+def utility_get_verbosity():
```

Comment:
Duplicated from test_validation.py -- move to common module and import e.g. `tests.utils`